src/sparseml/transformers/finetune/data/data_helpers.py

```diff
@@ -131,3 +137,71 @@ def make_dataset_splits(
         "calibration": calib_split,
     }
     return split_datasets
+
+
+def get_custom_datasets_from_path(path: str, ext: str = "json") -> Dict[str, str]:
+    """
+    Get a dictionary of custom datasets from a directory path. Support HF's load_dataset
+     for local folder datasets https://huggingface.co/docs/datasets/loading
+
+    This function scans the specified directory path for files with a
+     specific extension (default is '.json').
+    It constructs a dictionary where the keys are either subdirectory names or
+     direct dataset names (depending on the directory structure)
+    and the values are either file paths (if only one file exists with that name) or
+     lists of file paths (if multiple files exist).
+
+    :param path: The path to the directory containing the dataset files.
+    :param ext: The file extension to filter files by. Default is 'json'.
+
+    :return: A dictionary mapping dataset names to their file paths or lists of
+     file paths.
+
+    Example:
+        dataset = get_custom_datasets_from_path("/path/to/dataset/directory", "json")
+
+    Note:
+        If datasets are organized in subdirectories, the function constructs the
+         dictionary with lists of file paths.
+        If datasets are found directly in the main directory, they are included with
+         their respective names.
+
+    Accepts:
+        - path\
+            train.json
+            test.json
+            val.json
+
+        - path\
+            train\
+                data1.json
+                data2.json
+                ...
+            test\
+                ...
+            val\
+                ...
+
+    """
+    data_files = {}
+
+    if any(filename.endswith(ext) for filename in os.listdir(path)):
+        # If there are files with the given extension in the path
+        for filename in os.listdir(path):
+            if filename.endswith(ext):
+                name, _ = os.path.splitext(filename)
+                data_files[name] = os.path.join(path, filename)
```

Comment:
this could be simplified by using `Path.glob` look here: https://docs.python.org/3/library/pathlib.html

Reply:
I will add it on the followng pr!